scheduler-server/src/main/java/gov/nasa/jpl/aerie/scheduler/server/services/GenerateSchedulingLibAction.java

```diff
@@ -5,15 +5,20 @@
 import java.io.InputStreamReader;
 import java.util.Map;
 import java.util.Objects;
+import java.util.Optional;
 import java.util.stream.Collectors;
+
 import gov.nasa.jpl.aerie.scheduler.server.exceptions.NoSuchMissionModelException;
+import gov.nasa.jpl.aerie.scheduler.server.exceptions.NoSuchPlanException;
 import gov.nasa.jpl.aerie.scheduler.server.models.MissionModelId;
+import gov.nasa.jpl.aerie.scheduler.server.models.PlanId;
 
 public record GenerateSchedulingLibAction(
-    MissionModelService missionModelService
+    MissionModelService missionModelService,
+    PlanService.ReaderRole planService
 ) {
   public GenerateSchedulingLibAction {
-    Objects.requireNonNull(missionModelService);
+    Objects.requireNonNull(planService);
```

Comment:
I know we discussed potentially merging the missionModelService and PlanService since they're the same object, so this line may change, but could you clarify why `missionModelService` lost its `requireNonNull` check?